cli/lib/kontena/cli/stacks/upgrade_command.rb

```diff
@@ -21,13 +20,14 @@ class UpgradeCommand < Kontena::Command
     option '--[no-]deploy', :flag, 'Trigger deploy after upgrade', default: true
 
     option '--force', :flag, 'Force upgrade'
-    option '--skip-dependencies', :flag, "Do not install any stack dependencies"
+    option '--skip-dependencies', :flag, "Do not upgrade any stack dependencies", default: false
+    option '--reuse-values', :flag, "Reuse existing values"
```

Comment:
This looks like the `--use-defaults` flag in #2884

IIRC the situation was that the logic behaved correctly for `kontena stack upgrade`, but it went badly for `kontena stack install`.

The proposed logic in #2897 for `--no-prompt` would work for both `kontena stack upgrade` and also `kontena stack install`, including complex cases with dynamic resolvers as default values...